src/db/add_package.rs

```diff
@@ -1302,13 +1301,23 @@ mod test {
             let mut conn = env.async_db().async_conn().await;
 
             let crate_id = initialize_crate(&mut conn, "krate").await?;
-            let version: String = "version".repeat(100);
+            let version = Version::parse(&format!(
+                "1.2.3-{}+{}",
+                "prerelease".repeat(100),
+                "build".repeat(100)
+            ))?;
             let release_id = initialize_release(&mut conn, crate_id, &version).await?;
 
-            let db_version =
-                sqlx::query_scalar!("SELECT version FROM releases WHERE id = $1", release_id.0)
-                    .fetch_one(&mut *conn)
-                    .await?;
+            let db_version = sqlx::query_scalar!(
+                r#"
+                SELECT
+                    version as "version: Version"
```

Comment:
Won't that break for older crates that didn't have a valid version number?

Reply:
there aren't any old crates where the release version is broken. 

See also my comment here: https://github.com/Byron/crates-index-diff-rs/pull/53 

or [here on zulip](https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/does.20cargo.20guarantee.20package.20version.20is.20semver-parseable.3F/with/553277223 ). 

So , both cargo and crates.io guarantee it. 

Edge-cases exist, but only in the dependency `VersionReq` of 14 releases our of our ~2 mio :)